airbyte-cdk/java/airbyte-cdk/datastore-postgres/src/main/kotlin/io/airbyte/cdk/db/PostgresUtils.kt

```diff
@@ -18,7 +18,11 @@ object PostgresUtils {
         val jsonNodes =
             database.bufferedResultSetQuery(
                 { conn: Connection ->
-                    conn.createStatement().executeQuery("select * from pg_current_wal_lsn()")
+                    conn
+                        .createStatement()
+                        .executeQuery(
+                            "SELECT CASE WHEN pg_is_in_recovery() THEN pg_last_wal_receive_lsn() ELSE pg_current_wal_lsn() END AS pg_current_wal_lsn;"
```

Comment:
I have a question here:

Say that our database has no replica, and the db is in recovery mode, this would run into calling pg_last_wal_receive_lsn(), which will return null? I guess this would cause issue?